### PR TITLE
Intermediate explanation number two

### PR DESCRIPTION
**Pull request by MoonCWang**
Please note I could not find a way to anonymize or shorten my repl.it url.

- [x ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x ] My pull request targets the `master` branch of freeCodeCamp.
- [x ] None of my changes are plagiarized from another source without proper attribution.
- [x ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x ] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

File: guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md [hunk 1]
```diff
@@ -165,6 +165,40 @@ In the case that only one argument was passed, do not worry about how to prompt
...
+### Code Explanation:
+
+*   Change the argument object to an array to access array specific methods (documented below in first link)
+*   Continue if there are arguments and every argument is an integer 
+*   Curry if there is only one argument
+*   If prior fails, return undefined
+
+#### Relevant Links
+
+*   <a href='https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments' target='_blank' rel='nofollow'>arguments object</a>
```
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md]: Please change all these links to markdown format

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Check my comment

File: guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md [hunk 2]
```diff
@@ -165,6 +165,40 @@ In the case that only one argument was passed, do not worry about how to prompt
...
+    return function(y) {
+      if(Number.isInteger(y)){return arg[0] + y;}
+    }
+  } 
+  return undefined;
+}
+    // test here
+    console.log(JSON.stringify(addTogether(2,3)));
+```
+![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/@MoonCWang/addtogether' target='_blank' rel='nofollow'>Run Code</a>
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md]: @MoonCWang Also, please remove this line as we no longer want these in the challenge guide articles.

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See change request.

**Review by Manish-Giri — COMMENTED** [review]
Hi @MoonCWang 

Welcome to FreeCodeCamp, and Thank You for creating a Pull Request. 

However, the solution in your PR is inherently similar to the existing Advanced solution in the guide. It's best not to have two solutions with similar logic for the same challenge.

I will be closing this PR but feel free to create another one 👍

**Pull request closed without merging**